### PR TITLE
docs(checkbox): standalone page for checkbox documentation

### PR DESCRIPTION
**Pull request by MateoWartelle**
- [X] I have followed the [Your First Code Contribution section of the Contributing guide](https://github.com/themesberg/flowbite-react/blob/main/CONTRIBUTING.md#your-first-code-contribution)

Summarize the changes made and the motivation behind them.

To create separate pages for /forms/* documentations to match flowbite.

I had tried this previously but the branch was too far gone to continue (merge conflicts).

Reference related issues using `#` followed by the issue number.

If there are breaking API changes - like adding or removing props, or changing the structure of the theme - describe them, and provide steps to update existing code.

### CODE REVIEW

**Comment by MateoWartelle** [comment]
werid.. I can't tell what is wrong with the preview. The build passes locally. 

` ✓ Generating static pages (43/43) 
 ✓ Collecting build traces    
 ✓ Finalizing page optimization    

Route (app)                              Size     First Load JS
┌ ○ /                                    1.93 kB         217 kB
├ ○ /_not-found                          962 B            92 kB
└ ● /docs/[[...slug]]                    30 kB           214 kB
    ├ /docs/components/accordion
    ├ /docs/components/alert
    ├ /docs/components/avatar
    └ [+36 more paths]
+ First Load JS shared by all            91.1 kB
  ├ chunks/472-48e99746bd29f5a8.js       34.7 kB
  ├ chunks/fd9d1056-4e50bd5b75371e6d.js  53.8 kB
  ├ chunks/main-app-657973b50a7bf24c.js  282 B
  └ chunks/webpack-c7746c740a8989b6.js   2.3 kB


○  (Static)  prerendered as static HTML
●  (SSG)     prerendered as static HTML (uses getStaticProps)

✨  Done in 24.49s.`

**Comment by MateoWartelle** [comment]
@rluders can you check and Ill fix it? I can't tell why the preview failed.

File: data/docs-sidebar.ts [hunk 1]
```diff
@@ -70,7 +70,10 @@ export const DOCS_SIDEBAR: DocsSidebarSection[] = [
   {
     title: 'forms',
     href: '/forms/',
-    items: [{ title: 'Floating Label', href: '/docs/forms/floating-label', isNew: true }],
+    items: [
+      { title: 'Checkbox', href: '/docs/forms/checkbox', isNew: false },
```
Comment (SutuSebastian) [inline data/docs-sidebar.ts:74]: `isNew: false` is redundant since its an optional property

**Review by SutuSebastian — APPROVED** [review]
this is nice!

File: examples/checkbox/checkbox.listGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      language: 'tsx',
+      code,
+    },
+    {
+      fileName: 'server',
+      language: 'tsx',
+      code: codeRSC,
+    },
+  ],
+  githubSlug: 'checkbox/checkbox.listgroup.tsx',
```
Comment (SutuSebastian) [inline examples/checkbox/checkbox.listGroup.tsx:154]: rename `listgroup` to `listGroup`

File: examples/checkbox/index.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export { root } from './checkbox.root';
+export { disabled } from './checkbox.disabled';
+export { link } from './checkbox.link';
+export { helper } from './checkbox.helper';
+export { bordered } from './checkbox.bordered';
+export { listGroup } from './checkbox.listgroup';
```
Comment (SutuSebastian) [inline examples/checkbox/index.ts:6]: rename `listgroup` to `listGroup`

**Comment by SutuSebastian** [comment]
> weird.. I can't tell what is wrong with the preview. The build passes locally.
> 
> ```
> ✓ Generating static pages (43/43) 
> ✓ Collecting build traces    
> ✓ Finalizing page optimization    
> 
> Route (app)                              Size     First Load JS
> ┌ ○ /                                    1.93 kB         217 kB
> ├ ○ /_not-found                          962 B            92 kB
> └ ● /docs/[[...slug]]                    30 kB           214 kB
>     ├ /docs/components/accordion
>     ├ /docs/components/alert
>     ├ /docs/components/avatar
>     └ [+36 more paths]
> + First Load JS shared by all            91.1 kB
>   ├ chunks/472-48e99746bd29f5a8.js       34.7 kB
>   ├ chunks/fd9d1056-4e50bd5b75371e6d.js  53.8 kB
>   ├ chunks/main-app-657973b50a7bf24c.js  282 B
>   └ chunks/webpack-c7746c740a8989b6.js   2.3 kB
> 
> 
> ○  (Static)  prerendered as static HTML
> ●  (SSG)     prerendered as static HTML (uses getStaticProps)
> 
> ✨  Done in 24.49s.
> ```

This passes locally for u probably because u renamed the file afterwards from `listgroup` (lowercase) to `listGroup` (camelcase) and the IDE and git did not catch it (it happens quite often, don't worry).

Build fails due to:
```
./examples/checkbox/index.ts
Module not found: Can't resolve './checkbox.listgroup'
```

I left comments where it should be fixed.

**Comment by MateoWartelle** [comment]
> > weird.. I can't tell what is wrong with the preview. The build passes locally.
> > ```
> > ✓ Generating static pages (43/43) 
> > ✓ Collecting build traces    
> > ✓ Finalizing page optimization    
> > 
> > Route (app)                              Size     First Load JS
> > ┌ ○ /                                    1.93 kB         217 kB
> > ├ ○ /_not-found                          962 B            92 kB
> > └ ● /docs/[[...slug]]                    30 kB           214 kB
> >     ├ /docs/components/accordion
> >     ├ /docs/components/alert
> >     ├ /docs/components/avatar
> >     └ [+36 more paths]
> > + First Load JS shared by all            91.1 kB
> >   ├ chunks/472-48e99746bd29f5a8.js       34.7 kB
> >   ├ chunks/fd9d1056-4e50bd5b75371e6d.js  53.8 kB
> >   ├ chunks/main-app-657973b50a7bf24c.js  282 B
> >   └ chunks/webpack-c7746c740a8989b6.js   2.3 kB
> > 
> > 
> > ○  (Static)  prerendered as static HTML
> > ●  (SSG)     prerendered as static HTML (uses getStaticProps)
> > 
> > ✨  Done in 24.49s.
> > ```
> 
> This passes locally for u probably because u renamed the file afterwards from `listgroup` (lowercase) to `listGroup` (camelcase) and the IDE and git did not catch it (it happens quite often, don't worry).
> 
> Build fails due to:
> 
> ```
> ./examples/checkbox/index.ts
> Module not found: Can't resolve './checkbox.listgroup'
> ```
> 
> I left comments where it should be fixed.

ahhh thank you.

**Comment by SutuSebastian** [comment]
maybe we should remove the existing `checkbox` docs and examples which is under `components/forms/*` as we slowly migrate all of them and eventually deprecate for good `components/forms/*`

**Comment by SutuSebastian** [comment]
the disabled state doesn't look like https://flowbite.com/docs/forms/checkbox/#disabled-state

react
Screenshot 2023-11-14 at 23 55 50

core
Screenshot 2023-11-14 at 23 55 55

**Review by SutuSebastian — CHANGES_REQUESTED** [review]
the disabled state doesn't look like https://flowbite.com/docs/forms/checkbox/#disabled-state

react
Screenshot 2023-11-14 at 23 55 50

core
Screenshot 2023-11-14 at 23 55 55

File: examples/checkbox/checkbox.horizontalListGroup.tsx
Comment (SutuSebastian) [inline examples/checkbox/checkbox.horizontalListGroup.tsx:1]: in the core library clicking the entire section triggers a checked/unchecked action.



https://github.com/themesberg/flowbite-react/assets/41998826/c47e46c1-6f17-445d-8475-ae86076e9d4b

File: examples/checkbox/checkbox.colors.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import { type CodeData } from '~/components/code-demo';
+import { Checkbox, Label } from '~/src';
+
+const code = `
```
Comment (SutuSebastian) [inline examples/checkbox/checkbox.colors.tsx:4]: code here is incorrect, it should be the one below in the function

File: examples/checkbox/checkbox.colors.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        <Label htmlFor="purple-checkbo">Purple</Label>
+      </div>
+      <div className="flex items-center gap-2">
+        <Checkbox id="teal-checkbox" color="teal" checked />
+        <Label htmlFor="teal-checkbo">Teal</Label>
+      </div>
+      <div className="flex items-center gap-2">
+        <Checkbox id="yellow-checkbox" color="yellow" checked />
+        <Label htmlFor="yellow-checkbo">Yellow</Label>
+      </div>
```
Comment (SutuSebastian) [inline examples/checkbox/checkbox.colors.tsx:66]: I think orange is missing

Comment (MateoWartelle) [reply examples/checkbox/checkbox.colors.tsx:66]: orange is not in FlowBiteColors ?

**Comment by MateoWartelle** [comment]
> the disabled state doesn't look like https://flowbite.com/docs/forms/checkbox/#disabled-state
> 
> react Screenshot 2023-11-14 at 23 55 50
> 
> core Screenshot 2023-11-14 at 23 55 55

ohh I see Label was not disabled.

File: examples/checkbox/checkbox.colors.tsx (hunk 5 above)
Comment (SutuSebastian) [reply examples/checkbox/checkbox.colors.tsx:90]: oh that's true, but why isn't it 🤔

Comment (rluders) [reply examples/checkbox/checkbox.colors.tsx:90]: There are all the flowbite colors:
https://flowbite.com/docs/customize/colors/

However, it extends the TailwindCSS... so it means that...
https://tailwindcss.com/docs/customizing-colors

☝️ these colors (if now overwriten) are also available.

Comment (SutuSebastian) [reply examples/checkbox/checkbox.colors.tsx:90]: I see, gotcha! 🤝

Comment (rluders) [reply examples/checkbox/checkbox.colors.tsx:90]: BTW, just one side note here if someone wants to create an issue for this (not asking to include on this PR), we should work to provide all colors to all the components, and improve the color interface `FlowbiteColors`, `TailwindColors`, `FlowbiteStateColors` (default, active, enable, disable)... I don't know. But, we need to create some standard interface that needs to be available to all the components. Needs some PoC/Proposal/Investigation, maybe not all components need all the colors, and in this case, we may need to find a way to inform it in the docs.

But as I said, I don't expect this to be fixed on this PR... But we need a very well-described issue for this. (ChatGPT?)

Comment (SutuSebastian) [reply examples/checkbox/checkbox.colors.tsx:90]: lets get to digging/brainstorming then

File: examples/checkbox/checkbox.horizontalListGroup.tsx
Comment (MateoWartelle) [reply examples/checkbox/checkbox.horizontalListGroup.tsx:1]: I see, ok i will fix.

**Comment by rluders** [comment]
@MateoWartelle @SutuSebastian I think that we are good to merge this one, right? Just need to rebase.

**Comment by SutuSebastian** [comment]
> @MateoWartelle @SutuSebastian I think that we are good to merge this one, right? Just need to rebase.

I think some things broke down after rebase:
- sidebar does not contain the `checkbox` item anymore

Screenshot 2023-11-24 at 10 53 14

**Comment by SutuSebastian** [comment]
> @MateoWartelle @SutuSebastian I think that we are good to merge this one, right? Just need to rebase.

There's still this issue that hasn't been resolved:

https://github.com/themesberg/flowbite-react/assets/41998826/c47e46c1-6f17-445d-8475-ae86076e9d4b

and it applies to both `Checkbox list group ` and `Horizontal list group `

**Comment by rluders** [comment]
I notice that when the checkbox is marked by default to be `checked` it cannot be `uncheked` by clicking on it. Is it just a documentation thing? Is this expected?

File: examples/checkbox/checkbox.disabled.tsx [hunk 6]
```diff
@@ -33,8 +33,8 @@ function Component() {
         <Label htmlFor="disabled-checkbox">Default checkbox</Label>
       </div>
       <div className="flex items-center gap-2">
-        <Checkbox id="disabled-checked-state-checkbox" disabled checked />
-        <Label htmlFor="disabled-checked-state-checkbox" disabled>Disabled checked</Label>
+        <Checkbox id="disabled-checked-state-checkbox" disabled defaultChecked />
+        <Label htmlFor="disabled-checked-state-checkbox" disabled>Disabled defaultChecked</Label>
```
Comment (SutuSebastian) [inline examples/checkbox/checkbox.disabled.tsx:37]: small mistake here, revert to `Disabled checked`

File: examples/checkbox/checkbox.disabled.tsx [hunk 7]
```diff
@@ -14,8 +14,8 @@ function Component() {
...
```
Comment (SutuSebastian) [inline examples/checkbox/checkbox.disabled.tsx:18]: revert to `Disabled checked`

**Review by SutuSebastian — CHANGES_REQUESTED** [review]

**Comment by SutuSebastian** [comment]
Some minor differences:

`Checkbox list group`

Core
Screenshot 2023-11-24 at 23 25 56

React
Screenshot 2023-11-24 at 23 26 10

`Horizontal list group`

Core
Screenshot 2023-11-24 at 23 27 04
4f">

`Inline layout`

Core
Screenshot 2023-11-24 at 23 28 11

React
Screenshot 2023-11-24 at 23 28 22

`Advanced layout `

Core
Screenshot 2023-11-24 at 23 37 00

React
Screenshot 2023-11-24 at 23 37 06

**Pull request closed without merging**